packages/core/src/components/dropdown-item/dropdown-item.scss

```diff
@@ -77,6 +77,7 @@ $focusBorder: 0.0625rem;
     overflow: hidden;
     text-overflow: ellipsis;
     white-space: nowrap;
+    padding-left: 0.25rem;
```

Comment:
Why not increase the margin of the icon instead so that the extra space will only get applied if the icon is actually present?